cmd/mapkubeapis/map.go

```diff
@@ -52,8 +52,11 @@ func newMapCmd(out io.Writer, args []string) *cobra.Command {
 		Long:         "Map release deprecated or removed Kubernetes APIs in-place",
 		SilenceUsage: true,
 		Args: func(cmd *cobra.Command, args []string) error {
-			if len(args) != 1 {
-				return errors.New("name of release to be mapped needs to be passed")
+			if len(args) == 0 {
+				cmd.Help()
+				os.Exit(1)
```

Comment:
The exit code for this should be `0`.

Reply:
@jsoref Usually 0 is a success. This is an error and should the exit value 1 then?

Reply:
Running a command with no arguments and getting help isn't an error.

```sh
$ cmd=helm; out=$(mktemp); err=$(mktemp); $cmd > $out 2> $err; echo $? $(cat $out | wc -l) $(cat $err | wc -l); rm $out $err
0 94 0
$ cmd=kubectl; out=$(mktemp); err=$(mktemp); $cmd > $out 2> $err; echo $? $(cat $out | wc -l) $(cat $err | wc -l); rm $out $err
0 66 0
$ cmd=docker; out=$(mktemp); err=$(mktemp); $cmd > $out 2> $err; echo $? $(cat $out | wc -l) $(cat $err | wc -l); rm $out $err
0 0 78
$ cmd=false; out=$(mktemp); err=$(mktemp); $cmd > $out 2> $err; echo $? $(cat $out | wc -l) $(cat $err | wc -l); rm $out $err
1 0 0
```

Reply:
Some commands do return error when run with no arguments, for instance `git`. I think since providing more than one argument is also an error, it makes sense for none to be an error as well.

Reply:
I am now having doubts that we should be showing the usage. Looking again @jsoref issue raised in #24, running just `helm` was used as an example. This should show usage as no command is used. However, in the plugin case, the command is given but the mandatory arg of release is not. This to me this is an error and usage should not be shown. What do you think @flowjo-shaunr  @jsoref ?

Reply:
The main way to learn how to use a thing is via usage. When I run top level commands without arguments, they suggest how to use them, including subcommands. When I run those subcommands without arguments, they suggest how to use them.

I can provide examples...

Reply:
Should this be the sole job of the `--help` flag?

Reply:
I think it's still okay to be printing full usage, even as a subcommand (plugin), since it is a separate tool that doesn't necessarily inherit any of the root behaviors of the `helm` command, such as any global flags. That could be the sole job of the `--help` flag, but personally I feel that it's more convenient as a user to just receive the full usage of the plugin if I run nothing, rather than receiving a message that tells me to then run `--help`

Reply:
Thanks for the great feedback @jsoref @flowjo-shaunr. Taking this on board, let's keep the output of usage when just the command run and the error code of 1 because it is essentially an error as the mandatory arg is not passed.

Hope this is ok to all.